example for binary data from HESS J0632+957

GernotMaier opened the pull request:
I've created a first example for the kind of changes I think are necessary for the binaries. Main example is in input/data/2014/2014ApJ...780..168A:

- tev-000030.yaml with orbital parameters and different spectral results
- tev-000030-1-sed.ecsv to tev-000030-6-sed.ecsv, containing the SEDs from 6 different measurements
- tev-000030-lc.ecsv contains two light curves, from HESS and VERITAS. Need to find a good solution here to distinguish between the two instruments

cdeil commented on input/data/2012/2012ApJ...754L..10A/tev-000030-lc.ecsv:
```diff
@@ -0,0 +1,26 @@
+# - {SED_TYPE: flux}
+# - UL_CONF: 0.99
+# - comments: |
+# MAGIC light curve
```
Some of your ECSV files can't be read by Astropy table:
https://travis-ci.org/gammapy/gamma-cat/builds/187298896

To reproduce, you can run `./make.py check` locally, or directly try to read your ECSV file with Astropy Table:
```python
from astropy.table import Table
Table.read('https://raw.githubusercontent.com/GernotMaier/gamma-cat/3ea707cf54e129a0ee5d057db5939e663446d34d/input/data/2012/2012ApJ...754L..10A/tev-000030-lc.ecsv', format='ascii.ecsv')
# Error
```
I didn't look in detail, but I think it might just be that you need to indent the comments field properly. This should be OK:
```
# - comments: |
#     MAGIC light curve
```

cdeil commented on input/data/2012/2012ApJ...754L..10A/tev-000030-lc.ecsv:
```diff
@@ -0,0 +1,26 @@
+# - data_type: lc
+# - source_id: 30
+# - source_name: HESS J0632+057
+# - paper_id: 2012ApJ...754L..10A
```
We've changed from `paper_id` -> `reference_id` everywhere recently, because not all datasets will be from papers. Please also do the change for all the new files you're adding here.

cdeil commented:
There's a problem.

In https://github.com/GernotMaier/gamma-cat/blob/3ea707cf54e129a0ee5d057db5939e663446d34d/input/data/2014/2014ApJ...780..168A/tev-000030.yaml you have multiple `spec` keys.

It's not possible to structure the data like this. It can't be read. Try this:
```
with open('input/data/2014/2014ApJ...780..168A/tev-000030.yaml') as fh:
    data = yaml.load(fh)
```

I haven't thought about your use case yet, how to store such complex data.
My suggestion would be that you structure it how you think best and propose that here as a starting point for discussion (or we merge and continue discussion in the coming months).
The first goal is to have the info in machine-readable format at all, finding good ways to structure it will take time and feedback by people that enter and use the data.

> tev-000030-lc.ecsv contains two light curves, from HESS and VERITAS. Need to find a good solution here to distinguish between the two instruments

Make it two separate files?

cdeil commented:
I see you've added new columns to `input/gammacat/gamma_cat_columns.yaml` to cover the binary case here. I'm not sure it'll be possible to expose the binary info in a good way in the catalog.

The idea is that there will be two data products:
1. a series of many files that are the full "data archive" or "data collection"
2. a catalog that's a simple table with one source per row that contains only a small part of the full data collection

IMO it would be better to concentrate on how to enter the binary data in `input` for now, and then how to expose it in `output` as a set of processed files. 
But if you want to add stuff to the catalog here already and think it's useful, OK, we can keep it.

GernotMaier commented:
I've made several updates and  ./make.py is running now without errors / warnings:

- added additional data fields for binaries to dataset_source_info.schema.yaml
- fixed all paper_id fields to reference_id
- fixed all comment fields giving errors (really surprised how picky yaml is about spaces)
- divided one of the light curve files into two (tev-000030-lc.ecsv  is now tev-000030-1-lc.ecsv  and tev-000030-2-lc.ecsv )
- fixed a few more mistakes I've made in the ecvs format (again, really surprised how 'unforgiving' the format is)

I see that the travis checks are still failing. Will look into that later (was really hoping that this is much simpler). 

I didn't know that input/gammacat/gamma_cat_columns.yaml defines the output format. I can reverse these changes again if necessary.

cdeil commented:
> (really surprised how picky yaml is about spaces)

YAML uses indentation to declare structure, like Python
The alternative is to have something like JSON which uses braces and brackets all over the place.
IMO YAML is much more human-friendly, but of course it's a matter of taste.

> fixed a few more mistakes I've made in the ecvs format (again, really surprised how 'unforgiving' the format is)

ECSV uses YAML for the header. So if you don't like YAML, you won't like ECSV. :-)
Other that that ... it's good if a format is strict and unforgiving on unclear input, no?

--

The remaining error on travis-ci 
https://travis-ci.org/gammapy/gamma-cat/builds/187776288#L635
is that there's a formatting error in `gamma_cat_columns.yaml`, i.e. it's not longer valid YAML.

cdeil commented on input/gammacat/gamma_cat_columns.yaml:
```diff
@@ -299,6 +299,41 @@
+  fmt:
+  unit:
+  description: Observatory name
+ name: fileindex
```
There's a missing minus here on this line `name: fileindex`.
That's the cause of the YAML parse error:
```
ParserError: while parsing a block collection
  in "a.yaml", line 2, column 1
expected <block end>, but found '<block mapping start>'
  in "a.yaml", line 331, column 2
```

GernotMaier commented:
Looks like the built on travis CI when now through without problems, although running locally './make.py --loglevel debug all' gives me the following error:

DEBUG:gammacat.lightcurve:Reading /Users/maierg/Experiments/Gamma/gamma-cat-GM/gamma-cat/input/data/2014/2014ApJ...780..168A/tev-000030-1-lc.ecsv
DEBUG:gammacat.lightcurve:Reading /Users/maierg/Experiments/Gamma/gamma-cat-GM/gamma-cat/input/data/2014/2014ApJ...780..168A/tev-000030-2-lc.ecsv
DEBUG:gammacat.lightcurve:Reading /Users/maierg/Experiments/Gamma/gamma-cat-GM/gamma-cat/input/data/2016/2016arXiv161005799S/tev-000137-lc.ecsv
DEBUG:gammacat.utils:Reading /Users/maierg/Experiments/Gamma/gamma-cat-GM/gamma-cat/input/gammacat/gamma_cat_dataset.yaml
INFO:gammacat.utils:Writing /Users/maierg/Experiments/Gamma/gamma-cat-GM/gamma-cat/docs/data/gammacat-sources.json
INFO:gammacat.utils:Writing /Users/maierg/Experiments/Gamma/gamma-cat-GM/gamma-cat/docs/data/gammacat-datasets.json
INFO:gammacat.output:Writing /Users/maierg/Experiments/Gamma/gamma-cat-GM/gamma-cat/docs/data/gammacat-datasets.ecsv
DEBUG:gammacat.output:Processing SED: /Users/maierg/Experiments/Gamma/gamma-cat-GM/gamma-cat/input/data/2005/2005A%26A...432L..25A/tev-000110-sed.ecsv
DEBUG:gammacat.sed:Validating /Users/maierg/Experiments/Gamma/gamma-cat-GM/gamma-cat/input/data/2005/2005A%26A...432L..25A/tev-000110-sed.ecsv
Traceback (most recent call last):
  File "./make.py", line 117, in <module>
    cli()
  File "/Users/maierg/anaconda/lib/python3.4/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Users/maierg/anaconda/lib/python3.4/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Users/maierg/anaconda/lib/python3.4/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/maierg/anaconda/lib/python3.4/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/maierg/anaconda/lib/python3.4/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/maierg/anaconda/lib/python3.4/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "./make.py", line 111, in make_all
    ctx.invoke(make_output)
  File "/Users/maierg/anaconda/lib/python3.4/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "./make.py", line 47, in make_output
    maker.make_all()
  File "/Users/maierg/Experiments/Gamma/gamma-cat-GM/gamma-cat/gammacat/output.py", line 102, in make_all
    self.make_sed_files()
  File "/Users/maierg/Experiments/Gamma/gamma-cat-GM/gamma-cat/gammacat/output.py", line 127, in make_sed_files
    path.parent.mkdir(parents=True, exist_ok=True)
TypeError: mkdir() got an unexpected keyword argument 'exist_ok'

I don't really understand what the problem is. Can you spot it?

cdeil commented on input/data/2014/2014ApJ...794L...1A/tev-000089-sed.ecsv:
```diff
@@ -13,7 +13,6 @@
 # - source_id: 89
 # - source_name: TeV J1641-463
 # - reference_id: 2014ApJ...794L...1A
-# - url: https://www.mpi-hd.mpg.de/hfm/HESS/pages/publications/auxiliary/auxinfo_HESSJ1641
```
Why did you remove that line?
It's good to have the URL where the data comes from, no?

In cases where I can't point to a URL where I got the data via email, I'm usually adding a comment line:
```
Data from <name> obtained <how> <when>
```
If you can and think it's useful, please add that info where you got the data from in a comment somewhere.

GernotMaier replied:
Happened accidentally - I used this file as a template. I reversed the change

cdeil commented on input/data/2016/2016arXiv161003751S/tev-000030-1-sed.ecsv:
```diff
@@ -0,0 +1,26 @@
+# ---
+# datatype:
+# - {name: e_ref, datatype: float32}
+# - {name: e_lo, datatype: float32}
```
Looking at the values below, I'm pretty sure the columns should be `e_min` and `e_max` instead of `e_lo` and `e_hi` as you put here in the SED
See spec: http://gamma-astro-data-formats.readthedocs.io/en/latest/results/flux_points/index.html#energy-columns
and here for how we process energy columns:
https://github.com/gammapy/gamma-cat/blob/master/gammacat/sed.py#L67

I think this needs to be fixed in all (or at least several) SED files.

cdeil commented:
> TypeError: mkdir() got an unexpected keyword argument 'exist_ok'

The problem is that you're on Python 3.4 and I'm using a feature what was added in Python 3.5 in the scripts:

https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir
> Changed in version 3.5: The exist_ok parameter was added.

My thinking with the Python scripts here was that only a handful of people will ever run them, and they can be asked to install Python 3.5 or later. But if you find that annoying, I can make it work on Python 3.4  of course with almost no effort. Should I? (I'd really prefer not to support Python 2, even if it's easy, to encourage people to move on.)

GernotMaier commented:
> Changed in version 3.5: The exist_ok parameter was added.

> My thinking with the Python scripts here was that only a handful of people will ever run them, and they can be asked to install Python 3.5 or later. But if you find that annoying, I can make it work on Python 3.4 of course with almost no effort. Should I? (I'd really prefer not to support Python 2, even if it's easy, to encourage people to move on.)

I am a total python novice and I am really not an expert at all to help you with that decision.

For me, 3.4 would at this point be easier. But I really just want to add data...

I didn't manage to get the gammapy tests ( python -c 'import gammapy; gammapy.test()') running with 3.5 (but no problems with 3.4) although this seems to be a beginner's problem (as I said, I am really a novice). I've attached my error report, sorry to bother you with 

python -c 'import gammapy; gammapy.test()'

============================= test session starts ==============================
platform darwin -- Python 3.5.1, pytest-2.8.3, py-1.4.30, pluggy-0.3.1

Running tests with gammapy version 0.6.dev3945.
Running tests in gammapy.

Date: 2017-01-01T18:17:07

Platform: Darwin-16.3.0-x86_64-i386-64bit

Executable: /Users/maierg/anaconda/envs/p35/bin/python

Full Python Version: 
3.5.1 |Continuum Analytics, Inc.| (default, Jun 15 2016, 16:14:02) 
[GCC 4.2.1 Compatible Apple LLVM 4.2 (clang-425.0.28)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.11.2
Scipy: 0.18.1
Matplotlib: 1.5.3
h5py: 2.6.0
Pandas: 0.19.2+0.g825876c.dirty
cython: 0.25.2
pandas: 0.19.2+0.g825876c.dirty
skimage: 0.12.3
sklearn: 0.18.1
uncertainties: 3.0.1
iminuit: 1.2
astropy: 1.3
sherpa: 4.8.2
gammapy: 0.6.dev3945
naima: 0.8
reproject: 0.3.1
photutils: 0.3
gwcs: 0.7
wcsaxes: 0.9
aplpy: 1.1.1
regions: 0.1
astroplan: 0.2
Using Astropy options: remote_data.

rootdir: /Users/maierg/Experiments/Gamma/gammapy, inifile: setup.cfg
collected 600 items / 16 errors

gammapy/_fermipy/tests/test_skymap.py s
gammapy/astro/population/tests/test_simulate.py .......
gammapy/astro/population/tests/test_spatial.py ....................ssssssssss..........ssssssssss
gammapy/astro/source/tests/test_pulsar.py ......
gammapy/astro/source/tests/test_pwn.py ..
gammapy/astro/source/tests/test_snr.py .....
gammapy/background/tests/test_energy_offset_array.py s........
gammapy/background/tests/test_fov.py .
gammapy/background/tests/test_fov_cube.py sXXsss
gammapy/background/tests/test_models.py ...ssss..sss
gammapy/background/tests/test_off_data_background_maker.py s
gammapy/background/tests/test_reflected.py ssss
gammapy/background/tests/test_ring.py ....
gammapy/catalog/tests/test_analysis.py .
gammapy/catalog/tests/test_core.py .......
gammapy/catalog/tests/test_fermi.py ssssssssssssssssssssssssss
gammapy/catalog/tests/test_gammacat.py ssssssssssssssssssss
gammapy/catalog/tests/test_hess.py ssssssssssssssss
gammapy/catalog/tests/test_others.py sss
gammapy/catalog/tests/test_registry.py sss..
gammapy/catalog/tests/test_snrcat.py s
gammapy/catalog/tests/test_utils.py ...
gammapy/catalog/tests/test_xmatch.py ...
gammapy/cube/tests/test_core.py ssssssssss....ss
gammapy/cube/tests/test_cube_pipe.py s
gammapy/cube/tests/test_exposure.py s
gammapy/cube/tests/test_images.py s..
gammapy/cube/tests/test_sherpa_.py s
gammapy/data/tests/test_data_manager.py s
gammapy/data/tests/test_data_store.py sssssssssss
gammapy/data/tests/test_event_list.py sssssss
gammapy/data/tests/test_gti.py ss
gammapy/data/tests/test_hdu_index_table.py ss
gammapy/data/tests/test_obs_group.py ss.
gammapy/data/tests/test_obs_stats.py sss
gammapy/data/tests/test_obs_summary.py ssssssssss
gammapy/data/tests/test_obs_table.py ...
gammapy/data/tests/test_observationlist.py s
gammapy/data/tests/test_observers.py .
gammapy/data/tests/test_pointing.py sssssssss
gammapy/data/tests/test_target.py s
gammapy/datasets/tests/test_fermi.py sssssssssssssss
gammapy/datasets/tests/test_load.py ss
gammapy/datasets/tests/test_make.py ...x
gammapy/image/models/tests/test_gauss.py ............
gammapy/image/models/tests/test_model.py .
gammapy/image/models/tests/test_shapes.py ......sss...sss
gammapy/image/models/tests/test_theta.py ....x
gammapy/image/tests/test_asmooth.py s
gammapy/image/tests/test_catalog.py .sss
gammapy/image/tests/test_core.py ssssssssssssssssssssssssssssss..........s.
gammapy/image/tests/test_lists.py .......
gammapy/image/tests/test_mask.py ......
gammapy/image/tests/test_measure.py .....
gammapy/image/tests/test_plotting.py ...
gammapy/image/tests/test_profile.py .ss
gammapy/image/tests/test_radial_profile.py .
gammapy/image/tests/test_utils.py x...s
gammapy/irf/tests/test_effective_area.py .ss.
gammapy/irf/tests/test_energy_dispersion.py .ss
gammapy/irf/tests/test_fits_prod.py ss
gammapy/irf/tests/test_psf_3d.py sss
gammapy/irf/tests/test_psf_analytical.py sssss
gammapy/irf/tests/test_psf_check.py s
gammapy/irf/tests/test_psf_core.py .
gammapy/irf/tests/test_psf_king.py sss
gammapy/irf/tests/test_psf_table.py ...sss
gammapy/spectrum/tests/test_butterfly.py ..
gammapy/spectrum/tests/test_core.py ............
gammapy/spectrum/tests/test_cosmic_ray.py .
gammapy/spectrum/tests/test_crab.py ....
gammapy/spectrum/tests/test_diffuse.py .
gammapy/spectrum/tests/test_energy_group.py sss
gammapy/spectrum/tests/test_extract.py ssss
gammapy/spectrum/tests/test_fit.py sssssssss
gammapy/spectrum/tests/test_flux_point.py .....ssssssssssssssssssssssssssssssssss
gammapy/spectrum/tests/test_isrf.py xx.
gammapy/spectrum/tests/test_models.py ..............s
gammapy/spectrum/tests/test_obs_group.py s
gammapy/spectrum/tests/test_observation.py sssss
gammapy/spectrum/tests/test_powerlaw.py x..x...
gammapy/spectrum/tests/test_results.py ssss
gammapy/spectrum/tests/test_sed.py xxxxss
gammapy/spectrum/tests/test_simulation.py ....
gammapy/spectrum/tests/test_utils.py ...
gammapy/stats/tests/test_data.py ..
gammapy/stats/tests/test_feldman_cousins.py ....
gammapy/stats/tests/test_fit_statistics.py XX..
gammapy/stats/tests/test_poisson.py .x
gammapy/stats/tests/test_significance.py .....
gammapy/stats/tests/test_utils.py .
gammapy/time/tests/test_exptest.py .
gammapy/time/tests/test_lightcurve.py ...
gammapy/time/tests/test_plot.py .s
gammapy/time/tests/test_simulate.py .
gammapy/utils/coordinates/tests/test_celestial.py .
gammapy/utils/coordinates/tests/test_other.py .
gammapy/utils/distributions/tests/test_general_random.py x
gammapy/utils/root/tests/test_convert.py sss
gammapy/utils/tests/test_array.py ..
gammapy/utils/tests/test_axis.py .
gammapy/utils/tests/test_energy.py ..s.
gammapy/utils/tests/test_modeling.py .ss
gammapy/utils/tests/test_random.py ...
gammapy/utils/tests/test_scripts.py .
gammapy/utils/tests/test_time.py ...
gammapy/utils/tests/test_wcs.py .

==================================== ERRORS ====================================
______________ ERROR collecting gammapy/detect/test_statistics.py ______________
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
______________ ERROR collecting gammapy/detect/tests/test_cwt.py _______________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
____________ ERROR collecting gammapy/detect/tests/test_iterfind.py ____________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
_____________ ERROR collecting gammapy/detect/tests/test_kernel.py _____________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
______________ ERROR collecting gammapy/detect/tests/test_lima.py ______________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
_________ ERROR collecting gammapy/detect/tests/test_matched_filter.py _________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
________ ERROR collecting gammapy/detect/tests/test_test_statistics.py _________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
______________ ERROR collecting gammapy/scripts/tests/test_all.py ______________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/scripts/__init__.py:22: in <module>
    from .image_ts import *
gammapy/scripts/image_ts.py:8: in <module>
    from ..detect import compute_ts_image_multiscale
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
_________ ERROR collecting gammapy/scripts/tests/test_catalog_query.py _________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/scripts/__init__.py:22: in <module>
    from .image_ts import *
gammapy/scripts/image_ts.py:8: in <module>
    from ..detect import compute_ts_image_multiscale
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
____________ ERROR collecting gammapy/scripts/tests/test_cta_irf.py ____________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/scripts/__init__.py:22: in <module>
    from .image_ts import *
gammapy/scripts/image_ts.py:8: in <module>
    from ..detect import compute_ts_image_multiscale
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
__________ ERROR collecting gammapy/scripts/tests/test_data_select.py __________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/scripts/__init__.py:22: in <module>
    from .image_ts import *
gammapy/scripts/image_ts.py:8: in <module>
    from ..detect import compute_ts_image_multiscale
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
___________ ERROR collecting gammapy/scripts/tests/test_image_bin.py ___________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/scripts/__init__.py:22: in <module>
    from .image_ts import *
gammapy/scripts/image_ts.py:8: in <module>
    from ..detect import compute_ts_image_multiscale
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
___________ ERROR collecting gammapy/scripts/tests/test_image_fit.py ___________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/scripts/__init__.py:22: in <module>
    from .image_ts import *
gammapy/scripts/image_ts.py:8: in <module>
    from ..detect import compute_ts_image_multiscale
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
__________ ERROR collecting gammapy/scripts/tests/test_image_pipe.py ___________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/scripts/__init__.py:22: in <module>
    from .image_ts import *
gammapy/scripts/image_ts.py:8: in <module>
    from ..detect import compute_ts_image_multiscale
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
___________ ERROR collecting gammapy/scripts/tests/test_image_ts.py ____________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/scripts/__init__.py:22: in <module>
    from .image_ts import *
gammapy/scripts/image_ts.py:8: in <module>
    from ..detect import compute_ts_image_multiscale
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
___________ ERROR collecting gammapy/scripts/tests/test_spectrum.py ____________
_pytest.runner:149: in __init__
    ???
_pytest.main:431: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.main:431: in <lambda>
    return self._memoizedcall('_collected', lambda: list(self.collect()))
_pytest.python:599: in collect
    self.session._fixturemanager.parsefactories(self)
_pytest.python:1962: in parsefactories
    holderobj = node_or_obj.obj
_pytest.python:350: in fget
    self._obj = obj = self._getobj()
_pytest.python:596: in _getobj
    return self._memoizedcall('_obj', self._importtestmodule)
_pytest.main:311: in _memoizedcall
    res = function()
_pytest.python:606: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
py._path.local:650: in pyimport
    ???
gammapy/scripts/__init__.py:22: in <module>
    from .image_ts import *
gammapy/scripts/image_ts.py:8: in <module>
    from ..detect import compute_ts_image_multiscale
gammapy/detect/__init__.py:9: in <module>
    from .test_statistics import *
_pytest.assertion.rewrite:171: in load_module
    ???
gammapy/detect/test_statistics.py:21: in <module>
    from ._test_statistics_cython import (_cash_cython, _amplitude_bounds_cython,
E   ImportError: No module named 'gammapy.detect._test_statistics_cython'
 259 passed, 324 skipped, 13 xfailed, 4 xpassed, 1 pytest-warnings, 16 error in 43.68 seconds

cdeil commented:
@GernotMaier - Thank you!

I'm merging this now.

If you were still planning to do further commits on this branch, apologies!
Please open new PRs with further additions or changes.

Just for the record: I do plan to look at this more closely in the future and possibly suggest or apply myself data format changes.
